mobile/src/main/java/ch/epfl/sdp/mobile/application/social/SocialFacade.kt

```diff
@@ -0,0 +1,46 @@
+package ch.epfl.sdp.mobile.application.social
+
+import ch.epfl.sdp.mobile.application.Profile
+import ch.epfl.sdp.mobile.application.ProfileDocument
+import ch.epfl.sdp.mobile.application.authentication.AuthenticatedUser
+import ch.epfl.sdp.mobile.infrastructure.persistence.auth.Auth
+import ch.epfl.sdp.mobile.infrastructure.persistence.auth.User
+import ch.epfl.sdp.mobile.infrastructure.persistence.store.DocumentEditScope
+import ch.epfl.sdp.mobile.infrastructure.persistence.store.Store
+import ch.epfl.sdp.mobile.infrastructure.persistence.store.arrayUnion
+import ch.epfl.sdp.mobile.infrastructure.persistence.store.asFlow
+import ch.epfl.sdp.mobile.ui.social.Person
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.firstOrNull
+import kotlinx.coroutines.flow.map
+
+/**
+ * An interface which represents all the endpoints and available features for social interaction for
+ * a user of the Pawnies application.
+ *
+ * @param auth the [Auth] instance which will be used to handle authentication.
+ * @param store the [Store] which is used to manage documents.
+ */
+class SocialFacade(private val auth: Auth, private val store: Store) {
+
+  fun search(text: String): Flow<List<Profile>> {
+    return store.collection("users").whereEquals("name", text).asFlow<ProfileDocument>().map {
+      it.mapNotNull { doc -> doc?.toProfile() }
+    }
+  }
+
+  suspend fun follow(followed: Person, scope: DocumentEditScope) {
+    store.collection("users").document(auth.currentUser.firstOrNull()?.uid ?: "").update { scope.arrayUnion("following", followed.uid) }
+  }
```

Comment:
If you need access to the current user, why not define this as a method on `AuthenticatedUser` ? Also, you should not pass a `DocumentEditScope`.

The implementation would then approximately be as follows :

```kotlin
suspend fun follow(profile: Profile) {
  store.collection("users").document(profile.uid).update { 
    arrayUnion("following", profile.uid)
  }
}
```

Reply:
Thank you for your feedback Alexandre! It has been noted.